command/login.go

```diff
@@ -141,11 +140,14 @@ func (l *LoginCommand) Run(args []string) int {
 		for _, authMethod := range authMethodList {
 			if authMethod.Default {
 				l.authMethodName = authMethod.Name
+				if l.authMethodType == "" {
+					l.authMethodType = authMethod.Type
+				}
```

Comment:
What happens if the user specifies the auth method type, but this does not match the default one found? I assume the login will fail further down the flow, and wonder if we either want to error on this, or overwrite the passed flag value?

Reply:
Right, it crossed my mind. I think it's fine to let it fail down the line, I don't think we should be overwriting things. Are you ok with this? or do you want to make a check here and fail quickly?

Reply:
ok maybe it's best to fail quickly. 
[3e7dab8](https://github.com/hashicorp/nomad/pull/15869/commits/3e7dab826d82bb438deab46137b52b45059a2740)